service-registry/registry-zero-config/src/main/java/org/apache/servicecomb/zeroconfig/ZeroConfigDiscovery.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.servicecomb.zeroconfig;
+
+import com.netflix.config.DynamicPropertyFactory;
+import java.util.Collection;
+import java.util.List;
+import org.apache.servicecomb.registry.api.Discovery;
+import org.apache.servicecomb.registry.api.registry.Microservice;
+import org.apache.servicecomb.registry.api.registry.MicroserviceInstance;
+import org.apache.servicecomb.registry.api.registry.MicroserviceInstances;
+import org.apache.servicecomb.zeroconfig.client.ZeroConfigClient;
+
+public class ZeroConfigDiscovery implements Discovery {
+
+  private static final String NAME = "zero-config discovery";
+  private static final String ENABLED = "servicecomb.zeroconfig.registry.discovery.enabled";
```

Comment:
For zero config, `Discovery` and `Registration` must be enabled/disable togather. So it's better to use `servicecomb.zeroconfig.registry.enabled`. I'll modify other modules later.

Reply:
will just define this variable once in the Constant class/interface